core/services/registrysyncer/syncer.go

```diff
@@ -187,11 +186,16 @@ func (s *registrySyncer) updateStateLoop() {
 				// channel has been closed, terminating.
 				return
 			}
-			s.dbMu.Lock()
-			if err := s.orm.addState(ctx, *localRegistry); err != nil {
+			if err := s.orm.addLocalRegistry(ctx, *localRegistry); err != nil {
 				s.lggr.Errorw("failed to save state to local registry", "error", err)
 			}
-			s.dbMu.Unlock()
+			go func() {
+				select {
+				case <-s.stopCh:
+					return
+				case s.testUpdateChan <- true:
```

Comment:
@vyzaldysanchez This is going to leak a goroutine in production --

can we instead not spin up the goroutine and just have a default clause:

```
select {
case s.testUpdateChan <- true:
default:
}
```?

That way if testUpdateChan is uninitialized we'll just skip sending on the channel?